### PR TITLE
Improve client initialization documentation

### LINKED ISSUE #168: For CouchDB development startup options are confusing
If I start up the library with an opts object:

```
var opts = {
  account: 'myusername',
  password: 'mypassword',
  url: 'http://localhost:5984'
};
var cloudant = require('cloudant')(opts);
```

The library configures to point to `https://myusername:mypassword@myusername.cloudant.com` when the user may have expected to be pointing to `http://myusername:mypassword@localhost:5984`.

Raised by @bradnoble.

### PR DESCRIPTION
**Pull request by smithsz**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://github.com/cloudant/nodejs-cloudant/blob/master/DCO1.1.txt)
- [x] You have added tests for any code changes
- [x] You have updated the [CHANGES.md](https://github.com/cloudant/nodejs-cloudant/blob/master/CHANGES.md)
- [x] You have completed the PR template below:

## What
Improve client initialization documentation.

## Issues
Fixes #168.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,17 @@ You can find a further CRUD example in the [example](https://github.com/cloudant
 
 ### Initialization
 
+#### 1. Using a URL:
+
+You can also initialize Cloudant with a URL:
```
Comment (ricellis) [inline README.md]: This is first in the list so probably doesn't need to say also.

File: README.md [hunk 2]
```diff
@@ -129,6 +129,17 @@ You can find a further CRUD example in the [example](https://github.com/cloudant
...
+
+~~~ js
+var Cloudant = require('cloudant')
+var cloudant = Cloudant("http://MYUSERNAME:MYPASSWORD@localhost:5984");
+~~~
+
+#### 2. Using Cloudant account credentials:
+
 To use Cloudant, `require('cloudant')` in your code. That will return the initialization function. Run that function, passing your account name and password, and an optional callback. (And see the [security note](#security-note) about placing your password into your source code.
```
Comment (ricellis) [inline README.md]: Think this bit needs to move before the numbers:
> To use Cloudant, `require('cloudant')` in your code. That will return the initialization function. Run that function

and maybe add "with one of the following options" and then the "and an optional callback".

File: README.md [hunk 3]
```diff
@@ -138,14 +149,39 @@ var Cloudant = require('cloudant');
...
+Cloudant({account:me, username:otherUsername, password:otherPassword}, function(er, cloudant, reply) {
+  if (er) {
+    throw er;
+  }
+
+  console.log('Connected with username: %s', reply.userCtx.name);
+});
+~~~
+
+#### 3. Using Cloudant Local account credentials:
```
Comment (ricellis) [inline README.md]: Should this be a sub-section of URL?

File: README.md [hunk 4]
```diff
@@ -129,20 +129,24 @@ You can find a further CRUD example in the [example](https://github.com/cloudant
...
-#### 2. Using Cloudant account credentials:
+In general, the common style is that `Cloudant` (upper-case) is the **package** you load; whereas `cloudant` (lower-case) is your connection to your database--the result of calling `Cloudant()`:
+
+#### 2. Using account credentials:
 
-To use Cloudant, `require('cloudant')` in your code. That will return the initialization function. Run that function, passing your account name and password, and an optional callback. (And see the [security note](#security-note) about placing your password into your source code.
+##### 2.1. Connecting to Cloudant
 
-In general, the common style is that `Cloudant` (upper-case) is the **package** you load; whereas `cloudant` (lower-case) is your connection to your database--the result of calling `Cloudant()`:
+Connecting to Cloudant? You can just pass your account name and password (see the [security note](#security-note) about placing your password into your source code).
```
Comment (emlaver) [inline README.md]: Seems redundant to ask 'Connecting to Cloudant' under the same section

File: README.md [hunk 5]
```diff
@@ -129,20 +129,24 @@ You can find a further CRUD example in the [example](https://github.com/cloudant
 
 ### Initialization
 
+To use Cloudant, `require('cloudant')` in your code. Then initialize your client in one of the following ways:
```
Comment (emlaver) [inline README.md]: Do you need an 'add', 'use' or some verb? i.e. To use Cloudant, add require('cloudant') to your code

**Pull request merged**